token/js/test/unit/transferHook.test.ts

```diff
@@ -0,0 +1,139 @@
+import { getExtraAccountMetas, resolveExtraAccountMeta } from '../../src';
+import { expect } from 'chai';
+import { AccountMeta, Keypair, PublicKey } from '@solana/web3.js';
+
+describe('transferHookExtraAccounts', () => {
+    const testProgramId = new PublicKey('7N4HggYEJAtCLJdnHGCtFqfxcB5rhQCsQTze3ftYstVj');
+    const instructionData = Buffer.from(Array.from(Array(32).keys()));
+    const plainAccount = new PublicKey('6c5q79ccBTWvZTEx3JkdHThtMa2eALba5bfvHGf8kA2c');
+    const pdaPublicKey = new PublicKey('8Zibh9ywkKCZc3h23vwxvdMPG9yhg4LYYzky2yRkLHBJ');
+    const pdaPublicKeyWithProgramId = new PublicKey('9Apejge9XvJ3toZ8PX2sKKgMoyN7sSegzdq8n1kMzRae');
+
+    const plainSeed = Buffer.concat([
+        Buffer.from([1]), // u8 discriminator
+        Buffer.from([4]), // u8 length
+        Buffer.from('seed'), // 4 bytes seed
+    ]);
+
+    const instructionDataSeed = Buffer.concat([
+        Buffer.from([2]), // u8 discriminator
+        Buffer.from([4]), // u8 offset
+        Buffer.from([4]), // u8 length
+    ]);
+
+    const accountKeySeed = Buffer.concat([
+        Buffer.from([3]), // u8 discriminator
+        Buffer.from([0]), // u8 index
+    ]);
+
+    const addressConfig = Buffer.concat([plainSeed, instructionDataSeed, accountKeySeed], 32);
+
+    const plainExtraAccountMeta = {
+        discriminator: 0,
+        addressConfig: plainAccount.toBuffer(),
+        isSigner: false,
+        isWritable: false,
+    };
+    const plainExtraAccount = Buffer.concat([
+        Buffer.from([0]), // u8 discriminator
+        plainAccount.toBuffer(), // 32 bytes address
+        Buffer.from([0]), // bool isSigner
+        Buffer.from([0]), // bool isWritable
+    ]);
+
+    const pdaExtraAccountMeta = {
+        discriminator: 1,
+        addressConfig,
+        isSigner: true,
+        isWritable: false,
+    };
+    const pdaExtraAccount = Buffer.concat([
+        Buffer.from([1]), // u8 discriminator
+        addressConfig, // 32 bytes address config
+        Buffer.from([1]), // bool isSigner
+        Buffer.from([0]), // bool isWritable
+    ]);
+
+    const pdaExtraAccountMetaWithProgramId = {
+        discriminator: 128,
+        addressConfig,
+        isSigner: false,
+        isWritable: true,
+    };
+    const pdaExtraAccountWithProgramId = Buffer.concat([
+        Buffer.from([128]), // u8 discriminator
+        addressConfig, // 32 bytes address config
+        Buffer.from([0]), // bool isSigner
+        Buffer.from([1]), // bool isWritable
+    ]);
+
+    const extraAccountList = Buffer.concat([
+        Buffer.from([3, 0, 0, 0]), // u32 count
```

Comment:
The TLV library is using little-endian format for `u32` length, so you'll want to make sure your tests _and_ your buffer layouts are using little-endian.

https://github.com/solana-labs/solana-program-library/blob/737a37a52850a4990c73d16f5c3657844fc2c71a/libraries/type-length-value/src/pod.rs#L35-L48

Reply:
For some reason I always get confused with LE and BE. Isn't this already in LE?

```ts
const buffer = Buffer.from([ 3, 0, 0, 0 ])

buffer.readUInt32LE()
// 4

buffer.readUInt32BE()
// 67108864
```

Reply:
Ah shit, apparently I do too 😂. Looks like you're correct